runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -270,16 +267,50 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		}
 		else if (menuAction == MenuAction.EXAMINE_NPC && client.isKeyPressed(KeyCode.KC_SHIFT))
 		{
-			// Add tag option
+			// Add tag and tag-all options
+			final int id = event.getIdentifier();
+			final NPC[] cachedNPCs = client.getCachedNPCs();
+			final NPC npcCached = cachedNPCs[id];
+			boolean npcFound = true;
+			boolean matchesList = false;
+
+			if (npcCached == null || npcCached.getName() == null)
+			{
+				npcFound = false;
+			}
+			else
+			{
+				matchesList = highlights.stream()
+					.filter(highlight -> !highlight.equals(npcCached.getName().toLowerCase()))
```

Comment:
Declare `final String npcName = npc.getName().toLowerCase();` once and then reference it and below.